Update latest release section for 0.41.0 in website

Sreekala-Gopakumar opened the pull request:
https://github.com/eclipse-openj9/openj9-website/issues/358

Update latest release section for 0.41.0 in website

Closes #358
Signed-off-by: Sreekala Gopakumar <sreekala.gopakumar@ibm.com>

pshipton commented on index.html:
```diff
@@ -118,18 +118,25 @@ <h2>Join community calls</h2>
-  <li>Support for persistent shared classes cache is now added on z/OS&reg; systems.</li>
-  <li>A new <code>-XX:codecachetotalMaxRAMPercentage</code> option is added to set the maximum limit for the total JIT code cache size based on the amount of physical memory available to the VM.</li>
-  <li>Even though the container detection fails when you mount the <code>/proc</code> file system with the <code>hidepid=2</code> option on Linux® platforms, the VM starts but with a warning message.</li>
+  <li>The `-XX:+CompactStrings` option is now enabled by default on Java&trade; 11 and later.</li>
```
I don't think the backquotes work in html code.

pshipton replied:
I mean, you don't get any formatting change, you just see the backquotes on the web page.

pshipton commented on index.html:
```diff
@@ -118,18 +118,25 @@ <h2>Join community calls</h2>
+  <li>The `-XX:+CompactStrings` option is now enabled by default on Java&trade; 11 and later.</li>
+  <li>The JITServer client can now use the JITServer AOT cache even if both the `-Xshareclasses:readonly` option and the JITServer AOT cache feature were enabled at the same time. With the change in behavior of the `-Xshareclasses:readonly` option, the shared class cache startup creates a temporary new (writable) top layer that the JITServer AOT cache can use to store data that it needs to functi ...
+  <li>A new `-XX:[+|-]EnableDynamicAgentLoading` option is added to enable or disable the dynamic loading of agents into a running VM.</li>
+  <li>A new `-XX:[+|-]UseZlibNX` option is added to control the loading of the `zlibnx` library in the `LIBPATH` environment variable on the AIX systems.</li>
```
"on the AIX systems" could be "on AIX systems"

pshipton commented on index.html:
```diff
@@ -118,18 +118,25 @@ <h2>Join community calls</h2>
+<p><b>Performance highlights include:</b></p>
+<ul>
+  <li>Performance of algorithms is enhanced with the use of OpenSSL version 3 and later. Improved algorithms include SHA256, AES, HmacSHA256, and CHACHA20.</li>
+  <li>For Java 17, the performance to call the virtual or interface methods through `MethodHandles` is improved.</li>
```
I would change "to call the virtual or interface methods" to "to call virtual or interface methods"

Sreekala-Gopakumar commented:
@pshipton - Made the changes. Thanks!

Sreekala-Gopakumar commented:
@pshipton - I have updated the page. Please check and confirm. Thanks!

pshipton commented:
I think we can mention both "3.0.12, 3.1.4" since both these versions contain the fix. If we just mention 3.0.12, 3.1.0 may be considered a "later" version.

Sreekala-Gopakumar commented:
> I think we can mention both "3.0.12, 3.1.4" since both these versions contain the fix. If we just mention 3.0.12, then 3.1.0 may be considered a "later" version.

You mean instead of saying "... use OpenSSL version 3.0.12 or later... ", we should say "... use OpenSSL version 3.0.12 or 3.1.4..."?

pshipton commented:
Yes

Sreekala-Gopakumar commented:
Done

pshipton commented:
You removed "or later", which still applies. 3.0.12 or later, 3.1.3 or later.

Sreekala-Gopakumar commented:
> You removed "or later", which still applies. 3.0.12 or later, 3.1.4 or later.

This might be confusing. Which version is not applicable?

pshipton commented:
Discussed on Slack and suggested 3.0.12 or a later 3.0.x version, 3.1.4 or a later 3.1.x version.

The point being that either 3.0.x or 3.1.x can be used, but the required fix is introduced in 3.0.12 and 3.1.4.

Sreekala-Gopakumar commented:
Incorporated the suggested change. Please check and confirm. Thanks!

Sreekala-Gopakumar commented:
Updated the content based on the feedback received in https://github.com/eclipse-openj9/openj9-docs/pull/1200#issuecomment-1784392639

Sreekala-Gopakumar commented:
@pshipton - I have updated the release date. Added content related to com.sun.management.ThreadMXBean.getThreadAllocatedBytes() API support back ported from 0.42.release. Please check and confirm. Thanks!

pshipton commented on index.html:
```diff
@@ -118,18 +118,26 @@ <h2>Join community calls</h2>
+  <li>The JITServer client can now use the JITServer AOT cache even if both the <code>-Xshareclasses:readonly</code> option and the JITServer AOT cache feature were enabled at the same time. With the change in behavior of the <code>-Xshareclasses:readonly</code> option, the shared class cache startup creates a temporary new (writable) top layer that the JITServer AOT cache can use to store data t ...
+  <li>A new <code>-XX:[+|-]EnableDynamicAgentLoading</code> option is added to enable or disable the dynamic loading of agents into a running VM.</li>
+  <li>A new <code>-XX:[+|-]UseZlibNX</code> option is added to control the loading of the <code>zlibnx</code> library in the <code>LIBPATH</code> environment variable on AIX&reg; systems.</li>
+  <li>Support is added for the <code>com.sun.management.ThreadMXBean.getThreadAllocatedBytes()</code> API on all platforms except z/OS&reg;. The <code>isThreadAllocatedMemorySupported()</code>, <code>setThreadAllocatedMemoryEnabled()</code>, and <code>isThreadAllocatedMemoryEnabled()</code> methods are supported now.</li>
```
See my comment on the release notes about the methods being "supported now".

pshipton replied:
Pls indicate the z/OS support will be in the next release.

Sreekala-Gopakumar commented:
@pshipton - I have made the changes. Please check and confirm.

I have removed the details to maintain the consistency with the other points. Bullet point would have been too big. Anyways, we are mentioning in the next line  that "To read more about these and other changes, see the [OpenJ9 user documentation](https://www.eclipse.org/openj9/docs/openj9_releases/)." I hope that is ok. 

Thanks!

pshipton commented:
lgtm
